FDApy/representation/functional_data.py

```diff
@@ -0,0 +1,527 @@
+    def n_obs(self):
+        """Number of observations of the functional data.
+
+        Returns
+        -------
+        n_obs: int
+            Number of observations within the functional data.
+
+        """
+        return len(self.values)
+
+    @property
+    @abstractmethod
+    def range_obs(self):
+        pass
+
+    @property
+    def n_dim(self):
+        """Number of input dimension of the functional data.
+
+        Returns
+        -------
+        n_dim: int
+            Number of input dimension with the functional data.
+        """
+        return len(self.argvals)
+
+    @property
+    @abstractmethod
+    def range_dim(self):
+        pass
+
+    @property
+    @abstractmethod
+    def shape(self):
+        pass
+
+
+###############################################################################
+# Class DenseFunctionalData
+
+class DenseFunctionalData(FunctionalData):
+    r"""A class for defining Dense Functional Data.
+
+    A class used to define dense functional data. We denote by :math:`n`, the
+    number of observations and by :math:`p`, the number of input dimensions.
+    Here, we are in the case of univariate functional data, and so the output
+    dimension will be :math:`\mathbb{R}`.
+
+    Parameters
+    ----------
+    argvals: dict
+        The sampling points of the functional data. Each entry of the
+        dictionary represents an input dimension. The shape of the :math:`j`th
+        dimension is :math:`(m_j,)` for :math:`0 \leq j \leq p`.
+    values: np.ndarray
+        The values of the functional data. The shape of the array is
+        :math:`(n, m_1, \dots, m_p)`. It should not contain any missing values.
+
+    Examples
+    --------
+    >>> argvals = {'input_dim_0': np.array([1, 2, 3, 4]),
+                   'input_dim_1': np.array([5, 6, 7])}
+
+    >>> values = np.array([[[1, 2, 3], [1, 2, 3], [1, 2, 3], [1, 2, 3]],
+                   [[5, 6, 7], [5, 6, 7], [5, 6, 7], [5, 6, 7]],
+                   [[3, 4, 5], [3, 4, 5], [3, 4, 5], [3, 4, 5]],
+                   [[3, 4, 5], [3, 4, 5], [3, 4, 5], [3, 4, 5]],
+                   [[3, 4, 5], [3, 4, 5], [3, 4, 5], [3, 4, 5]]])
+
+    >>> DenseFunctionalData(argvals, values)
+    """
+
+    @staticmethod
+    def _check_argvals(argvals):
+        """Check the user provided `argvals`."""
+        FunctionalData._check_argvals(argvals)
+        _check_dict_type(argvals, np.ndarray)
+
+    @staticmethod
+    def _check_values(values):
+        """Check the use provided `values`."""
+        _check_type(values, np.ndarray)
+
+    @staticmethod
+    def _check_argvals_values(argvals, values):
+        """Check the compatibility of argvals and values."""
+        _check_dict_array(argvals, values)
+
+    def __init__(self, argvals, values):
+        """Initialize UnivariateFunctionalData object."""
+        super().__init__(argvals, values, 'univariate')
+
+    def __getitem__(self, index):
+        """Function call when self[index].
```

Comment:
Codacy found an issue: [First line should be in imperative mood; try rephrasing (found 'Function') (D401)](https://app.codacy.com/manual/StevenGolovkine/FDApy/pullRequest?prid=5929506)